Cars_Collision/cars_collision.cpp

```diff
@@ -0,0 +1,234 @@
+#include <iostream>
+#include <cmath>
+#include <vector>
+
+class Object 
+{
+    public:
+    double x_value, y_value;    
+    double velocity,angle;
+
+    Object()
+    {
+        x_value  = 0;
+        y_value  = 0;
+        velocity = 0;
+        angle    = 0;
+    }
+
+    Object(double x_value, double y_value, double velocity, double angle)
+    {
+        this -> x_value  = x_value;
+        this -> y_value  = y_value;
+        this -> velocity = velocity;
+        this -> angle    = angle;  
```

Comment:
No need to access variables like this inside the class you can also access by name directly
So you should follow the naming convention above that i mentioned. Than you will not have to right this -> 
everytime